src/main/java/org/jabref/gui/EntryFromIDViewModel.java

```diff
@@ -0,0 +1,228 @@
+                            basePanel.getDatabase().insertEntry(entry);
+                            break;
+                        case KEEP_BOTH:
+                            basePanel.getDatabase().insertEntry(entry);
+                            break;
+                        case KEEP_MERGE:
+                            basePanel.getDatabase().removeEntry(duplicate.get());
+                            basePanel.getDatabase().insertEntry(dialog.getMergedEntry());
+                            break;
+                        default:
+                            // Do nothing
+                            break;
+                    }
+                } else {
+                    // Regenerate CiteKey of imported BibEntry
+                    new BibtexKeyGenerator(basePanel.getBibDatabaseContext(), prefs.getBibtexKeyPatternPreferences()).generateAndSetKey(entry);
+                    basePanel.insertEntry(entry);
+                }
+                searchSuccesfulProperty.set(true);
+
+            } else if (StringUtil.isBlank(idText.getValue())) {
+                dialogService.showWarningDialogAndWait(Localization.lang("Empty search ID"), Localization.lang("The given search ID was empty."));
+            }
+            fetcherWorker = new FetcherWorker();
+
+            focusAndSelectAllProperty.set(true);
+            searchingProperty().setValue(false);
+
+        });
+    }
+
+    public void runFetcherWorkerForLookUp(Label lookUpField) {
+        searchSuccesfulProperty.set(false);
+        fetcherWorker.run();
+        fetcherWorker.setOnFailed(event -> {
+            Throwable exception = fetcherWorker.getException();
+            String fetcherExceptionMessage = exception.getMessage();
+            String fetcher = fetcherName;
+            String searchId = idText.getValue();
+            if (exception instanceof FetcherException) {
+                dialogService.showErrorDialogAndWait(Localization.lang("Error"), Localization.lang("Error while fetching from %0", fetcher + "." + "\n" + fetcherExceptionMessage));
+            } else {
+                dialogService.showErrorDialogAndWait(Localization.lang("No files found.", Localization.lang("Fetcher '%0' did not find an entry for id '%1'.", fetcher, searchId) + "\n" + fetcherExceptionMessage));
+            }
+            LOGGER.error(String.format("Exception during fetching when using fetcher '%s' with entry id '%s'.", searchId, fetcher), exception);
+
+            searchingProperty.set(false);
+
+            fetcherWorker = new FetcherWorker();
+
+        });
+
+        fetcherWorker.setOnSucceeded(evt -> {
+            Optional<BibEntry> result = fetcherWorker.getValue();
+
+            if (result.isPresent()) {
+
+                final BibEntry entry = result.get();
+
+                String entryString = "Found with : \n" + fetcherName + "\n\n with ID : \n" + entry.getId() + "\n\n Title : \n" + entry.getTitle() + "\n\n Publication date : \n" + entry.getPublicationDate();
```

Comment:
Please localize that. I think, wrapping in Localizaion.lang is the thing you are searching for.

Please no spaces before the `:`.